ci-operator/step-registry/rosa/cluster/provision/rosa-cluster-provision-commands.sh

```diff
@@ -223,33 +214,42 @@ if [[ "$ENABLE_BYOVPC" == "true" ]]; then
   fi
 fi
 
+# STS options
+STS_SWITCH="--non-sts"
+ACCOUNT_ROLES_SWITCH=""
+if [[ "$STS" == "true" ]]; then
+  STS_SWITCH="--sts"
+
+  # Account roles
+  ACCOUNT_ROLES_PREFIX=$(cat "${SHARED_DIR}/account-roles-prefix")
+  echo -e "Validate the ARNs of the account roles with the prefix ${ACCOUNT_ROLES_PREFIX}"
+  Account_Installer_Role_Name="${ACCOUNT_ROLES_PREFIX}-Installer-Role"
+  Account_ControlPlane_Role_Name="${ACCOUNT_ROLES_PREFIX}-ControlPlane-Role"
+  Account_Support_Role_Name="${ACCOUNT_ROLES_PREFIX}-Support-Role"
+  Account_Worker_Role_Name="${ACCOUNT_ROLES_PREFIX}-Worker-Role"
+
+  roleARNList=$(rosa list account-roles -o json | jq -r '.[].RoleARN')
+
+  Account_Installer_Role_ARN=$(echo "$roleARNList" | { grep "${Account_Installer_Role_Name}" || true; })
+  Account_ControlPlane_Role_ARN=$(echo "$roleARNList" | { grep "${Account_ControlPlane_Role_Name}" || true; })
+  Account_Support_Role_ARN=$(echo "$roleARNList" | { grep "${Account_Support_Role_Name}" || true; })
+  Account_Worker_Role_ARN=$(echo "$roleARNList" | { grep "${Account_Worker_Role_Name}" || true; })
+  if [[ -z "${Account_ControlPlane_Role_ARN}" ]] || [[ -z "${Account_Installer_Role_ARN}" ]] || [[ -z "${Account_Support_Role_ARN}" ]] || [[ -z "${Account_Worker_Role_ARN}" ]]; then
```

Comment:
There is no 'ControlPlane' for hosted-cp cluster. The related flag is still can be set. One potential problem is that no ControlPlane role created if use the `rosa create account-roles --hosted-cp` command.

Reply:
First time to know that, let us talk about this later.